src/damavand/cloud/aws/resources/vllm_component.py

```diff
@@ -125,19 +125,29 @@ def __init__(
         )
 
         self.args = args
+
         _ = self.model
         _ = self.endpoint_config
         _ = self.endpoint
 
-        if self.args.public_internet_access:
-            _ = self.api
-            _ = self.api_resource_completions
-            _ = self.api_method
-            _ = self.api_integration
-            _ = self.api_integration_response
-            _ = self.api_method_response
-            _ = self.api_deployment
-            _ = self.endpoint_ssm_parameter
+        _ = self.api
+        _ = self.api_resource_v1
+        _ = self.api_resource_chat
+        _ = self.api_resource_completions
+
+        # Only create API key if public internet access is set to False
+        if not self.args.public_internet_access:
```

Comment:
It looks like you are changing the behavior to always create the API Gateway which I do agree with but, change the arg name to something meaningful and related like this:

```suggestion
        if not self.args.is_api_key_required:
```